packages/forms-web-app/src/dynamic-forms/dynamic-components/number-entry/question.js

```diff
@@ -39,6 +39,17 @@ class NumberEntryQuestion extends Question {
 	prepQuestionForRendering(section, journey, customViewData, payload) {
 		let viewModel = super.prepQuestionForRendering(section, journey, customViewData, payload);
 
+		const answer = journey.response.answers[this.fieldName];
+		const persistedAnswer =
+			answer !== undefined && answer !== null && answer === 0
+				? `${answer}`
+				: answer !== undefined && answer !== null
+				? answer
+				: '';
```

Comment:
as it was agreed to let the number of witnesses question have a minimum of 0 this had a lot of strange behaviours everywhere as 0 is being treated as falsy. 

I couldnt figure out how to get the persistence on the question page to work for 0 without making it a string and all other numbers stay as numbers..

Reply:
If this is the logic we need, we should consider putting it into a simple utility function. That way you can use `if` and `return` instead of nested ternaries. 

e.g.

```
// or some sensible name
const asNumberAnswer(answer) => {
  if (answer === 0) {
    return answer.toString();
  }
  // etc...
}'
```

Reply:
> Wondering if it needs some more fundamental changes, is something in the dynamic display meant to be checking for null/undefined, but just checks for falsey?

Im thinking there is a check somewhere thats meant to be checking for null/ undefined but is checking falsey but I can find anything if it is this. 

Even if changing the question.value to be 0 which is what the nunjucks template is using it still doesnt display unless making it string. Again this makes me think its to do with a falsy check hiding somewhere.

Also realised the section wasnt being marked as complete in the section.js file as it was checking if there was no answer which 0 evaluated to, meaning the submit button wouldnt appear if all other answers were completed.

I'll add a util function